Fixes #674 - Update paid app submission test to add a payment account

bobsilverberg opened the pull request:
This will check for the presence of a payment account and create one if none exist. I did test this locally and it worked, but I am now unable to delete the payment account that I created, so I cannot test the logic that checks for no payment accounts again. Any reviewers will have the same issue.

@krupa Are you able to delete the payment account that was created when I ran my version of this, so reviewers can test it? I think that would need to be coordinated because after this code creates another payment account we would need intervention again to delete it to test again.

@rbillings | @m8ttyB | @davehunt r? please and thank you

bobsilverberg commented:
Adhoc running at http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/marketplace.adhoc/4/, although it doesn't prove much as the payment account now exists, so most of the new code isn't being run.

bobsilverberg commented:
This one is better: http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/marketplace.adhoc/5/

The tests need to be run serially

Note that I have submitted https://github.com/mozilla/marketplace-tests/pull/685 for the xpasses which are unrelated to this PR.

davehunt commented:
I've asked a question on the original issue. The code itself looks fine but I'd like to understand the reasons behind making this change.

bobsilverberg commented:
@davehunt I added a commit which now has all of the tests that submit a new app using a new user, rather than an existing user. I updated the test to create a paid app so it always adds a payment account. Everything seems to be passing for me.

r?

davehunt commented on tests/desktop/developer_hub/conftest.py:
```diff
@@ -6,7 +6,15 @@
+
+
+@pytest.fixture
+def login_new(mozwebqa, existing_user):
```
What's the difference between `login_existing` and `login_new`?

bobsilverberg replied:
Nothing, it seems. I guess maybe a commit never made it to my Github repo or something. When I was working on it `login_new` used a `new_user` fixture, which also seems to be absent, so that tests that use `login_new` use a new user created via FxAPom. I'll look into this and fix it.

bobsilverberg commented:
@davehunt I pushed a new commit that fixes the `login_new` thing. All tests pass for me locally and I've kicked off an adhoc at http://webqa-ci-staging1.qa.scl3.mozilla.com:8080/job/marketplace.adhoc/14/

davehunt commented on pages/desktop/developer_hub/compatibility_and_payments.py:
```diff
@@ -17,12 +17,39 @@ class CompatibilityAndPayments(Base):
 
+    @property
+    def payment_accounts_exist(self):
+        return self.is_element_present(*self._payment_accounts_drop_down_locator)
```
This property isn't being used, we should remove it.

davehunt commented:
r+ with comment addressed

bobsilverberg commented:
Thanks @davehunt. I addressed your last comment and squashed the commits. Merging!
